### DISCUSSION TITLE
192x192 Pushing the Limits with ESP32-S3

### DISCUSSION THREAD

**Discussion by Lukaswnd** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#542] [Show and tell]
Heyy
I recently bought some new panels and I wanted to try what the esp32s3 can handle. I am using a gerneric board with 16mb flash and 8mb psram. (I keep the framebuffer in the internal heap.) 
So I put all my panels together and made an 192x192 Display resulting in 36864 pixels. As you can see in the Images I'm usually using smaller combinations and just put them next to each other. Most of them are 64x32 P5 and the one in the left top is 64x32 P3.
I'm, using 5 bit colordeph and the Framebuffer gets shown with 35hz. with 1 transition bit 67hz would even be possible.

![WhatsApp Image 2023-12-16 at 13 46 58 (1)](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/28658448/d8e7ae6a-0d31-45df-a395-843cc877f433)
![WhatsApp Image 2023-12-16 at 13 46 58](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/28658448/243b6a61-9b24-4a4f-bb16-68b262157690)

I need to add, that if you stand at a larger distance, like 5 meters, the image looks really good. It's like the eye adds in the missing details you can't display. Unfortunately I could not catch this on picture.

I got the ESP32-S3 to it's limits, but I had to run a view tasks in the Backgound, like a WiFi Service, Webserver, NTP Time, Online Serial and a [berry](https://github.com/berry-lang/berry) interpreter. I tried to use the psram as much as possible, but arduino limits it. So I endet up with ~20k free bytes in the internal heap. Which can lead to 0, when you upload a file or too many webclients make requests. 
I think you could push it even further but if you want to use WiFi you should swtich from Arduino to esp-idf, to imporve memory handling etc. I think 256x256 would be possible, but then you are limited in wifi.

Since I am using berry to display stuff, I was not interested in high refresh rates or displaying gifs/videos, just in the momory usage.

So overall nice library and good job you guys are doing here!

**Comment by mrcodetastic** [comment]
Thanks for the kind words! Looks amazing as well.
